internal/translation/deployment/compare.go

```diff
@@ -61,10 +61,21 @@ func ComputeChanges(desired, current *Cluster) (*Cluster, bool) {
 		changes.DiskSizeGB = desired.DiskSizeGB
 	}
 
+	diskScaling := false
+
 	changesReplicationSpecs := make([]*akov2.AdvancedReplicationSpec, 0, len(desired.ReplicationSpecs))
 	for _, desiredReplicationSpec := range desired.ReplicationSpecs {
 		changesRegionConfig := make([]*akov2.AdvancedRegionConfig, 0, len(desiredReplicationSpec.RegionConfigs))
 		for _, desiredRegionConfig := range desiredReplicationSpec.RegionConfigs {
+			as := getAutoScalingChanges(desiredRegionConfig.AutoScaling)
+
+			scalingEnabled := as != nil &&
+				as.DiskGB != nil &&
+				as.DiskGB.Enabled != nil &&
+				*as.DiskGB.Enabled
+
+			diskScaling = diskScaling || scalingEnabled
```

Comment:
where else is `diskScaling` set between line 64 and here?
Otherwise, this could be simply:
```suggestion
			diskScaling = scalingEnabled
```

Reply:
because this is in a loop - we want this to be true if any of the region configs have autoscaling enabled.